scripts/us_census/pep/monthly_population_estimate/preprocess.py

```diff
@@ -0,0 +1,369 @@
+                            inplace=True)
+        self.df.drop_duplicates("Date", keep=keep_value, inplace=True)
+        self.df.drop(['date_range'], axis=1, inplace=True)
+        self.df.to_csv(self.cleaned_csv_file_path, index=False)
+
+    def _generate_mcf(self, df_cols: list) -> None:
+        """
+        This method generates MCF file w.r.t
+        dataframe headers and defined MCF template
+
+        Arguments:
+            df_cols (list) : List of DataFrame Columns
+
+        Returns:
+            None
+        """
+        mcf_template = ("Node: dcid:{}\n"
+                        "typeOf: dcs:StatisticalVariable\n"
+                        "populationType: dcs:Person{}{}{}\n"
+                        "statType: dcs:measuredValue\n"
+                        "measuredProperty: dcs:count\n")
+        mcf = ""
+        for col in df_cols:
+            residence = ""
+            status = ""
+            armedf = ""
+            if col.lower() in ["date", "location"]:
+                continue
+            if re.findall('Resident', col):
+                residence = "\nresidentStatus"
+                status = ": dcs:USResident"
+            if re.findall('ArmedForces', col):
+                residence = "\nresidentStatus"
+                if len(status) == 0:
+                    status = ": dcs:InUSArmedForcesOverseas"
+                else:
+                    status = ": dcs:USResident__InUSArmedForcesOverseas"
+            if re.findall('Resides', col):
+                if re.findall('Household', col):
+                    residence = "\nresidenceType"
+                    status = ": dcs:Household"
+            if re.findall('NonInstitutionalized', col):
+                residence = "\ninstitutionalization"
+                status = ": dcs:USC_NonInstitutionalized"
+            if re.findall('Civilian', col):
+                armedf = "\narmedForcesStatus: dcs:Civilian"
+            if re.findall('Count_Person_InUSArmedForcesOverseas', col):
+                armedf = "\narmedForcesStatus: dcs:InArmedForces"
+            mcf = mcf + mcf_template.format(col, residence, status,
+                                            armedf) + "\n"
+
+        # Writing Genereated MCF to local path.
+        with open(self.mcf_file_path, 'w+', encoding='utf-8') as f_out:
+            f_out.write(mcf.rstrip('\n'))
+
+    def _generate_tmcf(self, df_cols: list) -> None:
+        """
+        This method generates TMCF file w.r.t
+        dataframe headers and defined TMCF template
+
+        Arguments:
+            df_cols (list) : List of DataFrame Columns
+
+        Returns:
+            None
+        """
+        tmcf_template = ("Node: E:USA_Population_Count->E{}\n"
+                         "typeOf: dcs:StatVarObservation\n"
+                         "variableMeasured: dcs:{}\n"
+                         "measurementMethod: dcs:{}\n"
+                         "observationAbout: C:USA_Population_Count->Location\n"

```

Comment:
Can you change this based on MCF comment? (move to global, use named args)

Reply:
Made the code changes as per above suggestions.